webrecorder/browsertrix
# Limit URL list entry to maximum URLs

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/1222



### Changes
- Limits URL list entry to 1,000 URLs
- Limits additional URL list entry to 100 URLs
- Shows first invalid URL in list in error message
- Quick and dirty fix for long URLs wrapping: Show URLs in list on one line, with entire container scrolling

### Manual testing

1. Log in and go to Crawling
2. Click "New Workflow" -> "URL List"
3. Type in a valid URL and press "Enter". Verify that help text shows "1 URL entered"
4. Type in an invalid URL and press "Enter". Verify that help text shows invalid URL
5. Fix invalid URL and press "Enter". Verify help text updates to 2 urls
6. Repeat 3-5 and click outside of textarea instead of pressing "Enter". Verify validation updates as expected
7. Paste in 1,000 URLs (I generated them with a [mock data web tool](https://mockaroo.com/))
8. Enter in one more URL. Verify that help text updates to show max error
9. Go to review tab. Verify you're unable to save workflow with max error
10. Repeat with "Seeded Crawl" -> "Additional URLs", checking for max 100 instead of 1,000

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| New Workflow - URL List - Scope (updated info text) | <img width="865" alt="Screenshot 2023-10-02 at 7 58 37 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/2d247d38-0813-431a-a17b-285f80e8370d"> |
| New Workflow - URL List - Scope (invalid URL) | <img width="517" alt="Screenshot 2023-10-02 at 7 59 23 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/a4b49d7c-2952-4603-8c2f-0778e7907581"> |
| New Workflow - URL List - Scope (too many URLs) | <img width="521" alt="Screenshot 2023-10-02 at 7 58 53 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/ab8164e0-43ee-4ab3-9b7f-b77e1037e599"> |
